src/lib/compute_text_alternative.ts

```diff
@@ -17,49 +22,93 @@ import {rule2I} from './rule2I';
 export function computeTextAlternative(
   node: Node,
   context: Context = getDefaultContext()
-): string {
-  let result: string | null = '';
+): {name: string; nodesUsed: Set<Node>; rulesApplied: Set<Rule>} {
+  context.inherited.nodesUsed.add(node);
+  let result: string | null = null;
 
   result = rule2A(node, context);
   if (result !== null) {
-    return result;
+    context.inherited.rulesApplied.add('2A');
```

Comment:
It might be time to apply these rules in a loop rather than hard coding them? Hard-coding has resulted in a lot of repeated code here.

Reply:
That's true. I just had a play around with how to make this work and came up with [this](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgMIHtwQB5mQbwChllQALaYSAEwC5iTH8A3YAZyomoDl1qI2tZL34BtALoBuZCD4CAqmy5CAyhDAAeERAB80qAFcANgICCAB3NHgy5Gs0AlY7skBfSYVeFCMAyARgwJjIhiYATKYAFLL8QtoANMgImJC4QhhYuACUQmxgUKAA5gQMUOoGUCDIAOQw6OjVHl4+fgFBVaEQYQBC0XJxconJmWDpKThgOch5BSDFRCRlYBVV1QBGcFCNnt6EAPQAVMgAYsAm02CbYGzIFGXIB3veYACe5ihO5wC8NRHVyAAfX7dbaEYZ5ELOAAq6AAkgBbKxCfCiTqkKqfCDiIR9WLCQZJcZpNBEybIL46C6zQqucklEjVP5CToReIMRkg5nOHqeDzeXz+QLBZKIgyQKETUxGSCVOCBZgQXEQAb8IaksYjHIMEj4EBweHKqlFaQxBRKOh2dRaOR6SEmNgWKw2C32DSYnS0haEkYAOnIlBoPtNbEUXB9cGo1CVWQ8JDqUGQkXBeFRzkSwERRnEyHQMGQAHk1gArCABH0QcAFASRTowhFWLJZemMZMhATGPA-DNWJVqzWxxjAPM19vS5AAQi+PxAxiMTa9jG9qTAfpAd041B9nQdlmsYYjUc6Me1jCWK2bi5IeoNzNHYDZl8YwdDFuGy9X68Dz-ND8fdrMu7OkIb4TB+AZhtujp7tQv6Lu4J5eCQzSLOUlQXjI+qGtU1SwTIcghuawGkmBBRfvhL64ZBgG2CBuAkRuW7ODuTpcL+8GuEAA). What do you think?

Reply:
That's what I had in mind too! Added in next commit